### PR TITLE
Reduce number of `parse` calls

### LINKED ISSUE #39: CodeChecker parse command runs twice when changing between opening files
Let's assume that I have two opened files `a.cpp` and `b.cpp` and `a.cpp` is currently selected. Let's open the CodeChecker Output logs and now if I select the other opened file (`b.cpp`) in the logs I will see that the plugin will run the `CodeChecker parse` command twice:
```
>>> Starting process 'CodeChecker parse'
> /CodeChecker/bin/CodeChecker parse /helloworld/.codechecker -e json --file "/extension-output-codechecker.codechecker-#1"
>>> Process 'CodeChecker parse' exited with code 0
>>> Starting process 'CodeChecker parse'
> /CodeChecker/bin/CodeChecker parse /helloworld/.codechecker -e json --file "/extension-output-codechecker.codechecker-#1" "/helloworld/src/main.h"
>>> Process 'CodeChecker parse' exited with code 2
```

Also the problem is that the user will see the bugs only after the second command call is finished. This is a problem because if one command call takes 5 sec then the user have to wait 10 seconds.

### PR DESCRIPTION
**Pull request by Discookie**
Fixes #39.

The extension no longer executes a parse call when:
* Open files did not change
* There are no files open
* A changed window has 'output://' scheme - ref: #13

### CODE REVIEW

File: src/backend/processor/diagnostics.ts [hunk 1]
```diff
@@ -76,6 +76,13 @@ export class DiagnosticsApi {
             filesToLoad.push(this._selectedEntry.file);
         }
 
+        // Unload reports without calling parse
+        if (filesToLoad.length === 0) {
+            const dummyData: ParseResult = { reports: [], version: 1 };
+            this.parseDiagnosticsData(JSON.stringify(dummyData));
```
Comment (csordasmarton) [inline src/backend/processor/diagnostics.ts:82]: I don't really like this approach that we are creating a dummy data only to call this function. Why do we need to call `parseDiagnosticsData` here? To initialize `this._diagnosticEntries` and other variables? In this case we can initialize these fields here too. Also if it contains some extra logic we can create a separate function for it and we can call it here and in the `parseDiagnosticsData` function.

**Review by csordasmarton — CHANGES_REQUESTED** [review]

File: src/backend/processor/diagnostics.ts [hunk 2]
```diff
@@ -143,9 +150,28 @@ export class DiagnosticsApi {
...
+        }
+
+        for (const [idx, oldFile] of this._openedFiles.entries()) {
+            if (newFiles[idx] !== oldFile) {
+                this._openedFiles = newFiles;
+                this.reloadDiagnostics();
+
+                return;
+            }
+        }
```
[the remark this answers is not in the document]
Comment (Discookie) [reply src/backend/processor/diagnostics.ts]: Added, thanks!

File: src/backend/processor/diagnostics.ts (hunk 1 above)
Comment (Discookie) [reply src/backend/processor/diagnostics.ts]: Moved the field initializations here.

The parseMetadata does a little bit of extra processing when deciding whether this._selectedEntry should be kept, but selectedEntry cannot exist inside the unloading if, so it's no concern here.

**Review by csordasmarton — APPROVED** [review]
LGTM!

**Pull request merged**